includes/polling/os/allied.inc.php

```diff
@@ -24,21 +30,52 @@
     $version  = snmp_get($device, 'atiL2SwVersion.0', '-OsvQU', '+AtiL2-MIB');
 }
 
-// Allied Telesyn AT-8948 version 2.7.4-02 22-Aug-2005
-list($a,$b,$c,$d,$e,$f) = explode(' ', $device['sysDescr']);
+//Alliedware Plus 2.x.x.x 
+//Legacy products: 8100S
+//SNMPv2-MIB::sysDescr.0 = STRING: AlliedWare Plus (TM) 2.2.3.0
+if (!$hardware && !$version && !$features) {
+    $hardware = snmp_get($device, '.1.3.6.1.4.1.207.8.17.1.3.1.6.1', '-OsvQU', '+AT-SMI-MIB');
```

Comment:
Using the MIB here doesn't need the `+` in front of it and you should also switch to using the named OID and use snmp_get_multi_oid for these three together.

Reply:
- Ok have removed the `+` throughout the file. 
- I've done some analysis into the mibs, and I actually shouldn't have been using some of these mib files. A couple of them can translate an initial part of the OID, but not the whole thing. 
- As a result, I've removed reference to the mib, and kept the static OID. 

- I also don't *think* I can use snmp_get_multi_oid here. I tried the following:

```
if (!$hardware && !$version) {
$data2 = snmp_get_multi_oid($device, '.1.3.6.1.4.1.207.8.17.1.3.1.6.1 .1.3.6.1.4.1.207.8.17.1.3.1.5.1 .1.3.6.1.4.1.207.8.17.1.3.1.8.1', '-OsvQU');

$hardware = $data2['.1.3.6.1.4.1.207.8.17.1.3.1.6.1'];
$version = $data2['.1.3.6.1.4.1.207.8.17.1.3.1.5.1'];
$serial = $data2['.1.3.6.1.4.1.207.8.17.1.3.1.8.1'];
}
```

However, the box appears to get overloaded when polled (these are really low end CPU products and the CPU hits 100% when I run the poller.php):

```
SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 .1.3.6.1.4.1.207.8.17.1.3.1.6.1 .1.3.6.1.4.1.207.8.17.1.3.1.5.1 .1.3.6.1.4.1.207.8.17.1.3.1.8.1]
"AT-8100S/48POE"
"*"
"A04481G114700071"

Hardware: 
Version: 
Features: 
Serial: 

>> Runtime for poller module 'os': 0.4109 seconds with 4136 bytes
```


If I run the command manually, it works:

```
librenms@librenmsdev:~$ snmpget -v2c -c librenms -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied 10.37.105.93 .1.3.6.1.4.1.207.8.17.1.3.1.6.1 .1.3.6.1.4.1.207.8.17.1.3.1.5.1 .1.3.6.1.4.1.207.8.17.1.3.1.8.1
Bad operator (Settings): At line 41 in /opt/librenms/mibs/allied/AT-DOS-MIB
MIB search path: /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied
Cannot find module (FCMGMT-MIB): At line 28 in /opt/librenms/mibs/SW-MIB
Did not find 'connUnitPortStatEntry' in module #-1 (/opt/librenms/mibs/SW-MIB)
"AT-8100S/48POE"
"2.2.3.0"
"A04481G114700071"
```

And if I switch back to:

```
if (!$hardware && !$version) {
    $hardware = snmp_get($device, '.1.3.6.1.4.1.207.8.17.1.3.1.6.1', '-OsvQU');
    $version  = snmp_get($device, '.1.3.6.1.4.1.207.8.17.1.3.1.5.1', '-OsvQU');
    $serial   = snmp_get($device, '.1.3.6.1.4.1.207.8.17.1.3.1.8.1', '-OsvQU');
}
```
then I get:

```
#### Load poller module os ####
SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -m AtiSwitch-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 atiswitchProductType.0 atiswitchSwVersion.0 atiswitchSw.0]
No Such Object available on this agent at this OID
No Such Object available on this agent at this OID
No Such Object available on this agent at this OID

SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -m AtiL2-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 atiL2SwProduct.0]
No Such Object available on this agent at this OID

SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -m AtiL2-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 atiL2SwVersion.0]
No Such Object available on this agent at this OID

SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 .1.3.6.1.4.1.207.8.17.1.3.1.6.1]
"AT-8100S/48POE"

SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 .1.3.6.1.4.1.207.8.17.1.3.1.5.1]
"*"

SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied:/opt/librenms/mibs/allied udp:HOSTNAME:161 .1.3.6.1.4.1.207.8.17.1.3.1.8.1]
"A04481G114700071"

SQL[INSERT INTO `eventlog` (`host`,`device_id`,`reference`,`type`,`datetime`,`severity`,`message`,`username`)  VALUES ('19','19','NULL','system',NOW(),'3','OS Version -> Plus 2.2.3.0','')] 
SQL[INSERT INTO `eventlog` (`host`,`device_id`,`reference`,`type`,`datetime`,`severity`,`message`,`username`)  VALUES ('19','19','NULL','system',NOW(),'3','Hardware -> AT-8100S/48POE','')] 
SQL[INSERT INTO `eventlog` (`host`,`device_id`,`reference`,`type`,`datetime`,`severity`,`message`,`username`)  VALUES ('19','19','NULL','system',NOW(),'3','Serial -> A04481G114700071','')] 
Hardware: AT-8100S/48POE
Version: Plus 2.2.3.0
Features: 
Serial: A04481G114700071
```


Although it's possible I've done something wrong here...... I just tried to look at some of the other vendor.inc.php files to get an idea on what to do for snmp_get_multi_oid

Reply:
Ignore. murrant  pointed out I shouldn't be using  '-OsvQU' since I'm specifying the oid manually.  Works now